objects/src/assets/nonfungible.rs

```diff
@@ -26,6 +26,28 @@ const FAUCET_ID_POS: usize = 1;
 #[cfg_attr(feature = "serde", serde(transparent))]
 pub struct NonFungibleAsset(Word);
 
+impl PartialOrd for NonFungibleAsset {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        /// This wrapper allows us to use iterators builtin partial ord implementation.
+        /// This is much safer than attempting to do this correctly.
+        #[derive(PartialEq, Eq)]
+        struct Helper<'a>(&'a Felt);
+        impl PartialOrd for Helper<'_> {
+            fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+                self.0.inner().partial_cmp(&other.0.inner())
+            }
+        }
+
+        self.0.iter().map(Helper).partial_cmp(other.0.iter().map(Helper))
```

Comment:
We should derive `PartialOrd` and `Ord` for `Felt` if that's possible.

Reply:
I've been trying to avoid doing that because technically field elements do not inherently have an order. Their integer representation does have order - so, if we need to compare field elements, we usually first convert them to integers (i.e., via `felt.as_int()` method).

Reply:
For this specific case, I think converting `Word` into a `Digest` (which should be a zero-cost conversion) and then comparing digests (which implement `PartialOrd`, I believe), could be a simpler solution.

Reply:
Could you link me `Digest`? I'm struggling to find one that isn't the trait. I think technically its also not zero-cast because we're working with a reference here so it would copy/clone it?

Reply:
The struct is actually called `RpoDigest` and is defined in `miden-crypto` but in this crate we import it under `Digest` alias [here](https://github.com/0xPolygonMiden/miden-base/blob/main/objects/src/lib.rs#L32).